plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -3008,11 +3007,12 @@ public boolean delegateMaterializedViewRefreshToConnector(ConnectorSession sessi
         return hiveMaterializedViewMetadata.refreshMaterializedView(session, name);
     }
 
-    public static Optional<SchemaTableName> getSourceTableNameFromSystemTable(SchemaTableName tableName)
+    private Optional<SchemaTableName> getSourceTableNameFromSystemTable(SchemaTableName tableName)
     {
-        return Stream.of(SystemTableHandler.values())
-                .filter(handler -> handler.matches(tableName))
-                .map(handler -> handler.getSourceTableName(tableName))
+        return systemTableProviders.stream()
```

Comment:
Are there any other uses of `SystemTableHandler.values`?

Reply:
no